src/Bridge/Faker/Provider/EnumProvider.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the "elao/enum" package.
+ *
+ * Copyright (C) 2017 Elao
+ *
+ * @author Elao <contact@elao.com>
+ */
+
+namespace Elao\Enum\Bridge\Faker\Provider;
+
+use Elao\Enum\EnumInterface;
+
+class EnumProvider
+{
+    /**
+     * Enum mapping as an array with :
+     *
+     * - alias for Enum class as key
+     * - Enum FQCN as value
+     *
+     * Example :
+     *
+     * ```
+     * [
+     *   'civility' => Civility::class,
+     *   'gender'   => Gender::class,
+     * ]
+     * ```
+     *
+     * @var array $enumMapping
+     */
+    private $enumMapping = [];
+
+    public function __construct(array $enumMapping)
+    {
+        foreach ($enumMapping as $enumAlias => $enumClass) {
+            $this->ensureEnumClass($enumClass);
+            $this->enumMapping[$enumAlias] = $enumClass;
+        }
+    }
+
+    public function enum(): EnumInterface
+    {
+        // Unimplemented yet
+    }
+
+    public function randomEnum($enumAlias): EnumInterface
+    {
+        // Unimplemented yet
+    }
+
+    /**
+     * Make sure that $enumClass is a proper Enum class. Throws exception otherwise.
+     *
+     * @param string $enumClass
+     *
+     * @throws \InvalidArgumentException When $enumClass is not a class or is not a proper Enum
+     */
+    private function ensureEnumClass(string $enumClass)
+    {
+        if (!classExists($enumClass)) {
+            throw new \InvalidArgumentException("$enumClass class does not exist");
+        }
+
+        $reflectionEnum = new ReflectionClass($enumClass);
+
+        if (!$reflectionEnum->implementsInterface(EnumInterface::class)) {
```

Comment:
Rather than using reflection, you can use [`is_a`](http://php.net/manual/fr/function.is-a.php) with third argument set to `true`.